Gem/Code/Source/Components/NetworkMatchComponent.cpp

```diff
@@ -97,4 +178,22 @@ namespace MultiplayerSample
             EndRound();
         }
     }
+
+    void NetworkMatchComponentController::AssignPlayerIdentity(Multiplayer::NetEntityId playerEntity)
+    {
+        const Multiplayer::ConstNetworkEntityHandle entityHandle = Multiplayer::GetNetworkEntityManager()->GetEntity(playerEntity);
+        if (entityHandle.Exists())
+        {
+            if (PlayerIdentityComponent* identity = entityHandle.GetEntity()->FindComponent<PlayerIdentityComponent>())
+            {
+                identity->AssignPlayerName(PlayerNameString::format("Player %d", m_nextPlayerId));
```

Comment:
if we wanted to give the player a name they chose, would it be best to look for a property on the `PlayerIdentityComponent` of this playerEntity at this point? Assuming that component would be responsible for sourcing the name from the "player"

Reply:
I have not thought of the entire identity pipeline but at the simplest level, we can have PlayerIdentityComponent send an RPC with a player's chosen name and have it be updated on the server, using Autonomous -> Authority RPC.